pkgs/tools/security/gnupg/21.nix

```diff
@@ -29,7 +29,8 @@ stdenv.mkDerivation rec {
     sed -i 's,"libpcsclite\.so[^"]*","${pcsclite}/lib/libpcsclite.so",g' scd/scdaemon.c
   ''; #" fix Emacs syntax highlighting :-(
 
-  configureFlags = optional x11Support "--with-pinentry-pgm=${pinentry}/bin/pinentry";
+  gpgPinentryBinaryPath = pinentry.gpgPinentryBinaryPath or "bin/pinentry";
+  configureFlags = optional (pinentry != null) "--with-pinentry-pgm=${pinentry}/${gpgPinentryBinaryPath}";
```

Comment:
Shouldn't you check against guiSupport here, to not unnecessarily have dependencies on pinentry?